### PR TITLE
Log In and Log Out Updates 

### PR DESCRIPTION
**Pull request by BerryRahul**

### CODE REVIEW

**Review by sushimon — COMMENTED** [review]
Comments/concerns:
- Is `LogInRequestModel` supposed to be empty?
- Some classes do not follow naming conventions (e.g. the `logInViewTest` class).

**Review by sushimon — DISMISSED** [review]

File: src/main/java/log_in/use_case/LogInInteractor.java [hunk 1]
```diff
@@ -2,26 +2,30 @@
...
      *
      * @param access to retrieve stuff from the database
      * @param outputBoundary contains the result of this use case
      */
-    public LogInInteractor(LogInDsGateway access, LogInOutputBoundary outputBoundary, UserFactory userFactory){
+    public LogInInteractor(LogInDsGateway access, LogInOutputBoundary outputBoundary, UserFactory factory){
         this.access = access;
         this.outputBoundary = outputBoundary;
-        this.userFactory = userFactory;
+        this.userFactory = new UserFactory();
```
Comment (izumidayo) [inline src/main/java/log_in/use_case/LogInInteractor.java:28]: A factory is being passed as a parameter, do we want to use that or do we want to initialize a new one here?

File: src/main/java/log_in/use_case/LogInInteractor.java [hunk 2]
```diff
@@ -44,11 +48,23 @@ private boolean checkPassword(String email, String pass){
     }
 
     /**
-     * @param toSet email of user that logged in
+     * @param email email of user that logged in
+     * @param pass password of user that logged in
```
Comment (izumidayo) [inline src/main/java/log_in/use_case/LogInInteractor.java:52]: can we add success here as well?

File: src/main/java/log_in/use_case/LogInInteractor.java [hunk 3]
```diff
@@ -69,8 +85,7 @@ private LogInResponseModel logInCheck(LogInRequestModel requestModel) throws Use
             throw new UserException("Incorrect Password");
         }
         if (this.checkPassword(requestModel.getEmail(), requestModel.getPassword())){
-            User loggedIn = userFactory.readUser(access.getUser(true, requestModel.getEmail(), requestModel.getPassword()));
-            setCurrentUser(loggedIn);
+            setCurrentUser(true, requestModel.getEmail(), requestModel.getPassword());
```
Comment (izumidayo) [inline src/main/java/log_in/use_case/LogInInteractor.java:88]: not sure what is happening here?

File: src/main/java/log_out/interface_adapters/LogOutController.java [hunk 4]
```diff
@@ -10,14 +10,17 @@ public class LogOutController {
...
+     * @param inputBoundary input boundary for the use case
+     */
     public LogOutController(LogOutInputBoundary inputBoundary){
         this.interactor = inputBoundary;
     }
 
     //Initializes the logOut Use case
-    public void logOutInitializer(LogOutControllerData dataFromController){
-        LogOutRequestModel requestModel = new LogOutRequestModel(dataFromController.getLogOutRequest());
+    public void logOutInitializer(LogOutUserInputData dataFromController){
```
Comment (izumidayo) [inline src/main/java/log_out/interface_adapters/LogOutController.java:22]: Is the InputData that's being passed in here being used?

File: src/main/java/log_in/use_case/LogInInteractor.java [hunk 5]
```diff
@@ -22,10 +22,10 @@ public class LogInInteractor implements LogInInputBoundary {
      * @param access to retrieve stuff from the database
      * @param outputBoundary contains the result of this use case
      */
-    public LogInInteractor(LogInDsGateway access, LogInOutputBoundary outputBoundary, UserFactory userFactory){
+    public LogInInteractor(LogInDsGateway access, LogInOutputBoundary outputBoundary, UserFactory factory){
```
Comment (sushimon) [inline src/main/java/log_in/use_case/LogInInteractor.java:25]: What is the point of having `factory` as a parameter if you're not going to use it?

File: src/main/java/log_out/use_case/LogOutRequestModel.java [hunk 6]
```diff
@@ -5,9 +5,10 @@ public class LogOutRequestModel {
 
     /**
      * Initialize a log_out Request
-     * @param logOutRequest A string to indicate a logout request has been made
+     *  A string to indicate a logout request has been made
      */
-    public LogOutRequestModel(String logOutRequest){
+
+    public LogOutRequestModel(){
```
Comment (izumidayo) [inline src/main/java/log_out/use_case/LogOutRequestModel.java:11]: again, is this supposed to be empty?

**Review by sushimon — COMMENTED** [review]
See comment about `LogInInteractor`'s constructor

**Review by reganq — DISMISSED** [review]
Looks good

**Review by izumidayo — DISMISSED** [review]
Left a few comments but I'm going to approve for the sake of workflow!

File: src/main/java/log_in/use_case/LogInInteractor.java (hunk 5 above)
Comment (BerryRahul) [reply src/main/java/log_in/use_case/LogInInteractor.java:25]: ah nice catch, I forgot to assign it 
thanks

**Review by reganq — APPROVED** [review]
looks good

File: src/main/java/log_in/use_case/LogInInteractor.java (hunk 1 above)
Comment (BerryRahul) [reply src/main/java/log_in/use_case/LogInInteractor.java]: Yeah, forgot to assign it thanks

**Review by izumidayo — DISMISSED** [review]

**Review by sushimon — APPROVED** [review]

File: src/main/java/log_out/use_case/LogOutRequestModel.java (hunk 6 above)
Comment (BerryRahul) [reply src/main/java/log_out/use_case/LogOutRequestModel.java:11]: Yeah, for logging out theres not really any data that needs to be processed the parameter that was there before was never being used. It works the same without it

**Review by JohannLeidenfrost — APPROVED** [review]
Lgtm

**Review by reganq — APPROVED** [review]
looks good

**Review by izumidayo — DISMISSED** [review]

File: src/main/java/log_out/interface_adapters/LogOutController.java (hunk 4 above)
Comment (BerryRahul) [reply src/main/java/log_out/interface_adapters/LogOutController.java:22]: it was a mistake, thanks

**Review by izumidayo — APPROVED** [review]

**Review by JohannLeidenfrost — APPROVED** [review]
lgtm

**Pull request merged**